src/shared/selinux-util.c

```diff
@@ -436,7 +436,14 @@ int mac_selinux_get_create_label_from_exe(const char *exe, char **label) {
         if (sclass == 0)
                 return -ENOSYS;
 
-        return RET_NERRNO(security_compute_create_raw(mycon, fcon, sclass, label));
+        r = security_compute_create_raw(mycon, fcon, sclass, label);
+        if (r < 0) {
+                PROTECT_ERRNO;
```

Comment:
We usually do not use `PROTECT_ERRNO` like this.
Please use `r = -errno`:
```c
if (r < 0) {
        r = -errno;
        return mac_selinux_enforcing() ? r : -ENOSYS;
}
```

Reply:
why not just do:

```c
r = RET_NERRNO(security_compute_create_raw(…));
if (r < 0)
        …